mountpoint-s3-client/src/s3_crt_client/put_object.rs

```diff
@@ -109,6 +82,87 @@ impl S3CrtClient {
             state: S3PutObjectRequestState::CreatingMPU(mpu_created),
         })
     }
+
+    pub(super) async fn put_object_single<'a>(
+        &self,
+        bucket: &str,
+        key: &str,
+        params: &PutObjectParams,
+        contents: impl AsRef<[u8]> + Send + 'a,
+    ) -> ObjectClientResult<PutObjectResult, PutObjectError, S3RequestError> {
+        let span = request_span!(self.inner, "put_object_single", bucket, key);
+        let start_time = Instant::now();
+        let response_headers: Arc<Mutex<Option<Headers>>> = Default::default();
```

Comment:
It was a bit hard to understand why we need a lock here. Seems like we need the CRT to write response headers to this variable so we can extract the values later on a different thread. Probably worth adding a comment.